lib/core/request.js

```diff
@@ -320,7 +330,10 @@ function processHeader (request, key, val) {
     key.length === 10 &&
     key.toLowerCase() === 'connection'
   ) {
-    throw new InvalidArgumentError('invalid connection header')
+    if (val.toLowerCase() === 'close') {
+      request.reset = true
+    }
+    request.headers += `${key}: ${val}\r\n`
```

Comment:
We should not set the header here...

Reply:
Better to do it in `Client` or where exactly?

Reply:
Instead, just switching the value on the `reset` will do what's needed, and just check for either one of the `close` or `keep-alive` options. The settings usually come in the `keep-alive` header if custom behavior is desired. Thoughts? 🤔

Reply:
Just remove `request.headers += `${key}: ${val}\r\n`` and make sure val is either keep-alive or close.